### PR TITLE
[CU-868672hg5] Adds custom presets to image gen advance settings

### PR DESCRIPTION
**Pull request by manuelurenah**
Video overview -> https://t8459928.p.clickup-attachments.com/t8459928/e1544cd4-3a0d-4104-bb92-ae244522c673/screen-recording-2023-11-14-11%3A39.webm?view=open

image

### CODE REVIEW

**Comment by JustMaier** [comment]
Task linked: [CU-868672hg5 Add quick toggles to advanced options](https://app.clickup.com/t/868672hg5)

File: src/libs/form/components/NumberSlider.tsx [hunk 1]
```diff
@@ -56,43 +77,73 @@ export function NumberSlider({
...
   };
 
   useEffect(() => {
-    if (!focused) {
-      setValue(value);
+    if (!state.focused) {
+      setState((current) => ({ ...current, _value: value }));
     }
-  }, [value, precision]);
+  }, [value, precision, state.focused]);
```
Comment (lrojas94) [inline src/libs/form/components/NumberSlider.tsx:103]: This should pretty much cover the useEffect above, might wanna add the preset here too.

File: src/libs/form/components/NumberSlider.tsx [hunk 2]
```diff
@@ -32,21 +42,32 @@ export function NumberSlider({
...
 }: NumberSliderProps) {
   const { classes, cx } = useStyles();
   const numberRef = useRef<HTMLInputElement>(null);
-  const [focused, setFocused] = useState(false);
-  const [_value, setValue] = useState(value);
-  const [changeEndValue, setChangeEndValue] = useState<number>();
-  const [computedWidth, setComputedWidth] = useState<string>();
+  const [state, setState] = useState<State>({
+    focused: false,
+    _value: value,
```
Comment (lrojas94) [inline src/libs/form/components/NumberSlider.tsx:53]: There's no advange at using `_` now that you have it all under a state 🤷 Can be`value` imho

File: src/libs/form/components/NumberSlider.tsx [hunk 3]
```diff
@@ -56,43 +77,73 @@ export function NumberSlider({
...
+    onChange?.(state.changeEndValue);
+  }, [state.changeEndValue]);
 
   useEffect(() => {
     if (!numberRef.current) return;
-    setComputedWidth(getComputedWidth(numberRef.current, min, max, precision));
+    setState((current) => ({
+      ...current,
+      // Just to keep ts happy :shrug:
+      computedWidth: numberRef.current
```
Comment (lrojas94) [inline src/libs/form/components/NumberSlider.tsx:115]: There's a check for `!numberRef.current` above... The heck is TS complaining about 👀

File: src/libs/form/components/SelectWrapper.tsx [hunk 4]
```diff
@@ -47,16 +58,38 @@ export function SelectWrapper<T extends string | number>({
...
+            {label}
+            <PresetOptions
+              color="blue"
+              options={presets}
+              value={selectedPreset}
+              onChange={(value) => {
+                setSelectedPreset(value);
+                onChange?.(value as T);
+              }}
+            />
```
Comment (lrojas94) [inline src/libs/form/components/SelectWrapper.tsx:87]: You could make a `PresetsLabel` component out of this all in all

**Review by lrojas94 — APPROVED** [review]

File: src/libs/form/components/NumberSlider.tsx (hunk 3 above)
Comment (manuelurenah) [reply src/libs/form/components/NumberSlider.tsx:115]: I know, I know 😬 but it complains anyways

File: src/libs/form/components/SelectWrapper.tsx (hunk 4 above)
Comment (manuelurenah) [reply src/libs/form/components/SelectWrapper.tsx:87]: When we start implementing in other input components, then we definitely should 😛 for now I think it's fine the way it is

**Review by lrojas94 — APPROVED** [review]

**Pull request merged**